MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
+                bombLoad[selectedBombType]++;
+                curLoad += BombType.getBombCost(selectedBombType);
+            } else {
+                loopSafety++;
+            }
         }
 
-        // # of bombs is the unit's weight / (bomb cost * 5)
-        int numBombs = (int) Math.floor((entity.getWeight() - weightModifier) /
-                (BombType.getBombCost(bombIndex) * 5.0));
-        bombChoices[bombIndex] = numBombs;
+        // Oops, nothing left - rocket launchers are always popular
+        if (Arrays.stream(bombLoad).sum() == 0) {
+            bombLoad[BombType.B_RL] = bombUnits;
+            return bombLoad;
+        }
 
-        ((IBomber) entity).setBombChoices(bombChoices);
+        // Randomly replace advanced ordnance with rockets or HE, depending on force rating and
+        // air-air/ground preference
+
+        List<Integer> advancedOrdnance = Arrays.asList(
+                BombType.B_LG,
+                BombType.B_ARROW,
+                BombType.B_HOMING,
+                BombType.B_LAA,
+                BombType.B_AAA,
+                BombType.B_THUNDER,
+                BombType.B_FAE_SMALL,
+                BombType.B_FAE_LARGE,
+                BombType.B_AS,
+                BombType.B_ASEW
+        );
+
+        switch (quality) {
+            case IUnitRating.DRAGOON_ASTAR:
+            case IUnitRating.DRAGOON_A:
+                randomThreshold = 5;
+                break;
+            case IUnitRating.DRAGOON_B:
+                randomThreshold = 10;
+                break;
+            case IUnitRating.DRAGOON_C:
+                randomThreshold = 25;
+                break;
+            case IUnitRating.DRAGOON_D:
+                randomThreshold = 40;
+                break;
+            case IUnitRating.DRAGOON_F:
+                randomThreshold = 80;
+                break;
+            default:
+                randomThreshold = 99;
```

I'm wondering whether we should throw an exception here, instead of assigning a default value to `randomThreshold`. There shouldn't be any point in which quality isn't within the F-A* bounds, so having an exception thrown would mean we get alerted to the problem rather than it failing silently.

---

Currently the method is called correctly, but the quality parameter is a straight int type. So it could be called inappropriately with bad values (negative, large numbers, etc.).

So, using a
`throw new IllegalArgumentException("Unrecognized rating value: " + quality);`
statement, and let the thrown exception get handled upstream?

---

Yeah, my thinking is that if we throw an exception we know something has gone wrong; if we silently handle it with a default value the only way we know there was a bad call is if someone happens to notice an unusually large quantity of rockets being handed out.